parsec/serde/schema.py

```diff
@@ -99,17 +100,27 @@ def get_obj_type(self, obj):
 
     def _get_schema(self, type):
         if self._instantiated_schemas is None:
+            # Assert all schemas key are Enum:
+            if not all([isinstance(k, Enum) for k in self.type_schemas]):
```

Comment:
I always do `in self.types_schemas.keys()` (feels more explicit) but maybe it's kind of pointless...

Reply:
However I would do an even stronger check here:
```python
enum_types = {type(k) for k in self.type_schemas.keys()}
if len(enum_types) != 1 or not issubclass(enum_types[0], Enum):
    raise ValueError …
```
This way we also enforce the keys are all from the same enum ;-)

Reply:
that's better, fixed